test/unit/mongoose_push_telemetry_metrics_test.exs

```diff
@@ -10,6 +11,16 @@ defmodule MongoosePushTelemetryMetricsTest do
   setup do
     TestHelper.reload_app()
 
+    # TelemetryMetricsPrometheus starts asynchronously - we need to wait until metrics are
+    # registered, otherwise tests count run into race condition.
+    # Becasue of how TelemetryMetricsPrometheus startup is implemented right now, the following
+    # call alone will only return after the initialization (first one will be sucessful),
+    # but in case something would change in future we'll "wait" and retry until it returns
+    # at least one metric.
```

Comment:
```suggestion
    # TelemetryMetricsPrometheus starts asynchronously - we need to wait until metrics are
    # registered, otherwise tests could run into a race condition.
    # Because of how TelemetryMetricsPrometheus startup is implemented right now, the following
    # call alone will only return after the initialization (first one will be successful),
    # but in case something would change in future we'll "wait" and retry until it returns
    # at least one metric.
```